tools/update-schema.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+""" Update schema in iRODS from local filesystem.
+"""
+
+import argparse
+import base64
+import hashlib
+import subprocess
+import os
+import re
+import sys
+from typing import List, Tuple, Union
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("schemadirectory", help="Schema directory on local filesystem")
+    parser.add_argument("schemacollection", help="Schema collection in iRODS")
+    parser.add_argument("--default-resource", default="irodsResc",
+                        help="Resource to put new data objects on")
+    return parser.parse_args()
+
+
+def main() -> None:
+    args: argparse.Namespace = parse_args()
+    anything_changed: bool = False
+    anything_failed: bool = False
+    for filename in ["metadata.json", "uischema.json"]:
+        (sync_changed, sync_failed) = synchronize(args.schemadirectory, args.schemacollection, filename, args.default_resource)
+        anything_failed = anything_failed or sync_failed
+        anything_changed = anything_changed or sync_changed
+    if anything_changed:
+        print("Changed")
+    sys.exit(1 if anything_failed else 0)
+
+
+def data_object_exists(collection_name: str, dataobject_name: str) -> bool:
+    checkexist_command = ["ils", os.path.join(collection_name, dataobject_name)]
+    (checkexist_code, checkexist_stdout, checkexist_stderr) = run_command(checkexist_command)
+    return "does not exist" not in checkexist_stderr
+
+
+def get_checksum(path, checksumtype):
+    if checksumtype == "md5":
+        hsh = hashlib.md5()
+    elif checksumtype == "sha2":
+        hsh = hashlib.sha256()
+    else:
+        raise ValueError(f"Checksum type {checksumtype} not supported.")
+
+    f = open(path, 'rb')
```

Comment:
Ensure that the file is properly closed after reading

```suggestion
    with open(path, 'rb') as f:
        while True:
            chunk = f.read(8192)
            if chunk:
                hsh.update(chunk)
            else:
                break
```